### PR TITLE
Cleanup logging by using logrus

### PR DESCRIPTION
**Pull request by saschagrunert**
We're switching from go-kit/kit/log to sirupsen/logrus to not have the
need for passing the logger around.

Do you think this makes sense and we should switch to logrus for the rest
of the package, too?

### CODE REVIEW

File: cmd/release-notes/main.go [hunk 1]
```diff
@@ -173,6 +172,14 @@ func (o *options) BindFlags() *flag.FlagSet {
 	return flags
 }
 
+func envDefault(key, def string) string {
+	value := os.Getenv(key)
+	if len(value) == 0 {
+		return def
+	}
+	return value
+}
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply cmd/release-notes/main.go]: Makes sense, I changed it that way. :+1:

**Comment by justaugustus** [comment]
@saschagrunert -- Can you add a topic on one of the Release Engineering meetings to discuss?

I'd love us to use a consistent logging package across the repo, instead of per tool, which merits some discussion.

/hold

**Comment by saschagrunert** [comment]
> @saschagrunert -- Can you add a topic on one of the Release Engineering meetings to discuss?
> 
> I'd love us to use a consistent logging package across the repo, instead of per tool, which merits some discussion.
> 
> /hold

Yes sure, I added it to the agenda. :+1:

**Comment by saschagrunert** [comment]
As discussed in the [meeting](https://docs.google.com/document/d/16GqCjnEh86w8yADcrUylNoE1y1sqjIMYNC_gdk5WPSQ) yesterday, I did a short research regarding logging possibilities:

| Framework | Pro | Contra |
| - | - | - |
| log | a standard package | no log levels, no structured logging |
| [golang/glog](https://github.com/golang/glog) | simple, used by a lot of k8s projects: `k8s.io/autoscaler`, `k8s.io/cloud-provider-alibaba-cloud`, `k8s.io/contrib`, `k8s.io/dns`, `k8s.io/heapster`, `k8s.io/kube-deploy`, `k8s.io/minikube`, `k8s.io/node-problem-detector`, `k8s.io/publishing-bot`, `k8s.io/test-infra`, `sigs.k8s.io/aws-encryption-provider`, `sigs.k8s.io/descheduler`, `sigs.k8s.io/krew`, `sigs.k8s.io/kubebuilder-declarative-pattern` | no structured logging, no logging interception (like printing the line number) |
| [k8s.io/klog](https://github.com/kubernetes/klog) | seems standard in k8s, maintained fork of glog | ? |
| [sirupsen/logrus](https://github.com/sirupsen/logrus) | probably the most used logger out there, nice features like logging interception to add context to the logs (like `file:line`), used for example by: `k8s.io/kompose`, `k8s.io/kubeadm`, `k8s.io/test-infra` | ? |
| [k8s.io/klog](https://github.com/kubernetes/klog) | seems standard in k8s, maintained fork of glog | ? |
| [go-kit/kit/log](https://github.com/go-kit/kit/tree/master/log) | already a dependency in this repository | ? |

**Comment by justaugustus** [comment]
@saschagrunert -- My vote is for `logrus`. The projects using it today are closest in function to the utilities we're building in this repo.

**Comment by onlydole** [comment]
+1 for `logrus` - I've worked on a few projects with that as one of the dependencies and it worked quite well!

**Comment by saschagrunert** [comment]
Rebased on top of the latest mater branch. PTAL :)

File: .golangci.yml [hunk 2]
```diff
@@ -47,7 +47,6 @@ linters-settings:
   errcheck:
     check-type-assertions: true
     check-blank: true
-    exclude: .errcheck-excludes
```
Comment (saschagrunert) [inline .golangci.yml]: This is not needed any more :)

**Comment by saschagrunert** [comment]
> LGTM! Thanks @saschagrunert! Are you planning to update `krel ff` in a subsequent PR?
> 
> /lgtm

Yes, it's already prepared locally and I'm just waiting for this one to get in :grin:

**Comment by justaugustus** [comment]
Great work, @saschagrunert!
/lgtm
/approve
/hold cancel

**Comment by hoegaarden** [comment]
❤️ in general

> Yes, it's already prepared locally and I'm just waiting for this one to get in 😁

and ❤️ in specific! 😄

**Comment by saschagrunert** [comment]
If you want to look at the next step: https://github.com/kubernetes/release/pull/966 :)

**Pull request merged**